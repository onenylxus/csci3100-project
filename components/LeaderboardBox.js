// Import
import React from 'react';
import { Image, Text, View } from 'react-native';
import { Col, Grid } from 'react-native-easy-grid';
import Style from '../assets/style';

// Export Leaderboard box
export default function LeaderboardBox({ username, popularity }) {
  return (
    <View style={Style.postBox}>
      <View style={{ flexDirection: 'row' }}>
        <Grid>
          <Col style={{ flexDirection: 'row', marginVertical: 20 }}>
            <Text>#1</Text>
            <Image
              style={{
                width: 32,
                height: 32,
                marginHorizontal: 8,
                marginTop: 4,
                borderRadius: 28,
              }}
              source={require('../assets/images/profile.png')}
            />
            <View style={{ flexDirection: 'column' }}>
              <Text style={{ fontSize: 14, fontWeight: 'bold' }}>
                Username: {username}
              </Text>
              <Text>Popularity: {popularity}</Text>
            </View>
          </Col>
        </Grid>
      </View>
    </View>
  );
}
