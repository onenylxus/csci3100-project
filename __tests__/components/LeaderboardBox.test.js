// Import
import React from 'react';
import { render } from '@testing-library/react-native';

// Import target component
import LeaderboardBox from '../../components/LeaderboardBox';

// Run jest tests
describe('CommentBox', () => {
  let element;

  beforeEach(() => {

    // Render
    element = render(<LeaderboardBox />);
  });

  it('matches snapshot', () => {
    expect(element.toJSON()).toMatchSnapshot();
  });
});
