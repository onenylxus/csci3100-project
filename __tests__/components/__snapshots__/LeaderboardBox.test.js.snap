// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBox matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f8f8f8",
      "margin": 4,
      "width": "75%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "marginVertical": 20,
          }
        }
      >
        <Text>
          #1
        </Text>
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 28,
              "height": 32,
              "marginHorizontal": 8,
              "marginTop": 4,
              "width": 32,
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "bold",
              }
            }
          >
            username
          </Text>
          <Text>
            content
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
